srml/metadata/src/lib.rs

```diff
@@ -209,6 +209,30 @@ pub struct ModuleConstantMetadata {
 	pub documentation: DecodeDifferentArray<&'static str, StringBuf>,
 }
 
+/// All the metadata about a module error.
+#[derive(Clone, PartialEq, Eq, Encode)]
+#[cfg_attr(feature = "std", derive(Decode, Debug, Serialize))]
+pub struct ErrorMetadata {
+	pub name: DecodeDifferentStr,
+	pub documentation: DecodeDifferentArray<&'static str, StringBuf>,
+}
+
+/// All the metadata about errors in a module.
+pub trait ModuleErrorMetadata {
```

Comment:
I should be able to remove this. It will just make the macro implementation bit more complicated.
Maybe I should also change `pub errors: DFnA<ErrorMetadata>,` to `pub errors: ODFnA<ErrorMetadata>,` so no error type will be None?

Reply:
I think having an array is okay, needs less space.

Reply:
I tried to remove the trait but it makes the macro much more complicated, mainly to handle the `&'static str` case

Reply:
One of the failed attempt https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=339b2cdbd6747104f373b152fb21914d

Reply:
Yeah the trait is okay. Sorry for the noise.